lib/expressions.js

```diff
@@ -60,20 +60,87 @@ class OperationExpression {
   interpret(context) {
     const args = this.operands.map((x) => x.interpret(context));
     const operator = OPERATORS[this.operator];
-    if (!operator) throw new Error(`Unknown operator: ${operator}`);
+    if (!operator) throw new Error(`Unknown operator: ${this.operator}`);
     return operator(...args);
   }
 
   toExpression() {
     const list = this.operands.map((x) => x.toExpression());
-    return '(' + list.join(this.operator) + ')';
+    return `(${list.join(this.operator)})`;
   }
 
   toJavaScript() {
-    const parameters = Array.from(this.identifiers);
-    const header = '(' + parameters.join(', ') + ')';
-    return header + ' => ' + this.toExpression();
+    let header = '(';
+    for (const identifier of this.identifiers) {
+      if (header.length > 1) header += ', ';
+      header += identifier;
+    }
+    header += ')';
+    return `${header} => ${this.toExpression()}`;
+  }
+}
+
+class ConditionExpression {
+  constructor(clauses) {
+    this.identifiers = new Set();
+    this.clauses = clauses;
+
+    for (const { condition, consequents } of clauses) {
+      if (condition.identifiers !== undefined) {
+        for (const identifier of condition.identifiers) {
+          this.identifiers.add(identifier);
+        }
+      }
+
+      for (const consequent of consequents) {
+        if (consequent.identifiers !== undefined) {
+          for (const identifier of consequent.identifiers) {
+            this.identifiers.add(identifier);
+          }
+        }
+      }
+    }
+  }
+
+  // eslint-disable-next-line consistent-return
+  interpret(context) {
+    for (const { condition, consequents } of this.clauses) {
+      if (condition.interpret(context) !== false) {
+        return consequents.reduce(
+          (_, consequent) => consequent.interpret(context),
+          undefined,
+        );
+      }
+    }
+  }
```

Comment:
- I do not like disabled eslint rules, just make return consistent.
- `!== false` is not ok, see fix
- It is not ok to start reducing from undefined in most cases, it is also strange to use reduce with ignored accumulator arg
```suggestion
  interpret(context) {
    for (const { condition, consequents } of this.clauses) {
      if (condition.interpret(context)) {
        return consequents.reduce(
          (_, consequent) => consequent.interpret(context),
          undefined,
        );
      }
    }
  }
```

Reply:
We can't do `if(condition.interpret(context))` because in LISP condition is false only if it evaluates to `nil` while in JavaScript it can be any falsy value like non-positive number. E.g. (cond (0 x)) would be a true condition.

This is why I did `!== false` (`nil` is interpreted to `false` in our current solution). I've made this explicit in the latest commit.